### ISSUE TITLE
[ui-shell]: IBM Accessibility violation error

### ISSUE THREAD

**Issue by jc-corrales** [carbon-design-system/carbon-for-ibm-dotcom#10918]
### Description

The bx-side-nav-menu-item tag from the UI Shell component is throwing the following accessibility error:
Violation
```
The tabbable element's role 'none' is not a widget role
A tabbable element must have a valid widget role
```
It seems the problem can be fixed when I assign those tags the role="button".

### Component(s) impacted

UI Shell Component
bx-side-nav-menu-item tag

### Browser

Chrome, Microsoft Edge

### Carbon for IBM.com version

v1.31.0

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

Legal Entity Repository, can be found in storybook itself

### Package

@carbon/web-components

### CodeSandbox example

Don't have

### Steps to reproduce the issue (if applicable)

1. Open storybook
2. Open the UI shell component
3. Open one of the menus with dropdowns from the example
4. Run the IBM Equal Access Accessibility Checker
5. Identify errors that say "The tabbable element's role 'none' is not a widget role"

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
